rust/datafusion/src/optimizer/utils.rs

```diff
@@ -217,9 +217,133 @@ mod tests {
     use super::*;
     use crate::logicalplan::Expr;
     use arrow::datatypes::DataType;
+    use arrow::datatypes::DataType::*;
     use std::collections::HashSet;
     use std::sync::Arc;
 
+    #[test]
+    fn test_supertype_numeric_types() {
+        let types = vec![
+            UInt8, UInt16, UInt32, UInt64, Int8, Int16, Int32, Int64, Float32, Float64,
+        ];
+        let mut result = String::new();
+        for l in &types {
+            for r in &types {
+                result.push_str(&format!(
+                    "supertype of {:?} and {:?} is {:?}
+\n",
+                    l,
+                    r,
+                    get_supertype(l, r).unwrap()
+                ));
+            }
+        }
+        assert_eq!(
```

Comment:
Can we avoid comparing with string? whenever there's an error it is very hard to find out since you are comparing two huge strings.